src/Google.Storage.V1/UpdateObjectOptions.cs

```diff
@@ -0,0 +1,109 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using static Google.Apis.Storage.v1.ObjectsResource;
+using static Google.Apis.Storage.v1.ObjectsResource.UpdateRequest;
+
+namespace Google.Storage.V1
+{
+    /// <summary>
+    /// Options for <c>UpdateObject</c> operations.
+    /// </summary>
+    public class UpdateObjectOptions
+    {
+        /// <summary>
+        /// If present, selects a specific revision of this object (as opposed to the latest version, the default).
+        /// </summary>
+        public long? Generation { get; set; }
+
+        /// <summary>
+        /// Precondition for update: the object is only updated if the existing object's
+        /// generation matches the given value.
+        /// </summary>
+        public long? IfGenerationMatch { get; set; }
+
+        /// <summary>
+        /// Precondition for update: the object is only updated if the existing object's
+        /// generation does not match the given value.
+        /// </summary>
+        public long? IfGenerationNotMatch { get; set; }
+
+        /// <summary>
+        /// Precondition for update: the object is only updated if the existing object's
+        /// meta-generation matches the given value.
+        /// </summary>
+        public long? IfMetagenerationMatch { get; set; }
+
+        /// <summary>
+        /// Precondition for update: the object is only updated if the existing object's
+        /// meta-generation does not match the given value.
+        /// </summary>
+        public long? IfMetagenerationNotMatch { get; set; }
+
+        /// <summary>
+        /// The projection of the updated object to return.
+        /// </summary>
+        public Projection? Projection { get; set; }
+
+        /// <summary>
+        /// A pre-defined ACL for simple access control scenarios.
+        /// </summary>
+        public PredefinedAcl? PredefinedAcl { get; set; }
+
+        internal void ModifyRequest(UpdateRequest request)
+        {
+            // Note the use of ArgumentException here, as this will basically be the result of invalid
+            // options being passed to a public method.
+            if (IfGenerationMatch != null && IfGenerationNotMatch != null)
+            {
+                throw new ArgumentException($"Cannot specify {nameof(IfGenerationMatch)} and {nameof(IfGenerationNotMatch)} in the same options", "options");
+            }
+            if (IfMetagenerationMatch != null && IfMetagenerationNotMatch != null)
+            {
+                throw new ArgumentException($"Cannot specify {nameof(IfMetagenerationMatch)} and {nameof(IfMetagenerationNotMatch)} in the same options", "options");
+            }
+
+            if (Generation != null)
+            {
+                request.Generation = Generation;
```

Comment:
Generation.Value
(Actually, I think I'd argue that they should all be without the .Value)